subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -252,7 +253,8 @@ public PathVisitor(MerkleDirectorySnapshotBuilder builder, @Nullable Spec<FileTr
 
         @Override
         public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
-            String name = stringInterner.intern(dir.getFileName().toString());
+            String fileName = Optional.ofNullable(dir.getFileName()).map(Object::toString).orElse("");
```

Comment:
Do you have any idea how fast this `Optional` creating is as compared to doing a `null` check? The method here is not very hot (given that it happens only once per directory), so it probably does not matter.

Reply:
I hope it's inlined, lemme' check

Reply:
Created a [simple JMH](https://gist.github.com/paplorinc/231e71405158570322e0027a3f1aae6b#file-optionalbenchmark-java-L36) to validate and both seem fast enough:
```
Benchmark                     Mode  Cnt          Score         Error  Units
OptionalBenchmark.nullCheck  thrpt   20  107887111.061 ±  882182.482  ops/s
OptionalBenchmark.optional   thrpt   20   86746312.090 ± 1150860.296  ops/s
```